subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/client/TestExecutionClient.kt

```diff
@@ -18,27 +19,30 @@ class TestExecutionClient {
     private val requests: Channel<ClientTestRunRequest> = Channel(Channel.UNLIMITED)
     private val results: Channel<ClientTestRunResult> = Channel(Channel.UNLIMITED)
 
-    fun start(executionServiceCommunication: IntentionExecutionService.Communication): Communication {
-        // TODO: Don't use global scope. Unconfined coroutines lead to leaks
-        GlobalScope.launch {
-            for (request in requests) {
-                statesMapping[request.intention] = request.state
-                executionServiceCommunication.intentions.send(request.intention)
+    suspend fun start(
+        executionServiceCommunication: IntentionExecutionService.Communication,
+        scope: CoroutineScope
+    ): Communication {
+        scope.launch(Dispatchers.Unconfined) {
```

Comment:
Does it matter what to use when we only work with channels?

Reply:
You are right. I pretend that I understand coroutines